includes/MediaWiki/Api/EditEndpoint.php

```diff
@@ -141,17 +157,15 @@ public function run() {
 				$otherItem->getItem(),
 				'Reconciliation Edit',
 				EDIT_NEW,
-				// TODO actually do a token check?
-				false
+				$this->user->getEditToken()
```

Comment:
I don’t think this is the right way to do the token check – we’re getting the user’s token and then asking `EditEntity` to compare it to the user’s token, which should always succeed. The token should come from the request parameters, if I understand correctly.

Reply:
Hm, in master of MediaWiki core, `UpdateHandler` and `EditHandler::getActionModuleToken()` have some interesting related code – apparently session providers can declare whether they’re potentially vulnerable to CSRF or not, and depending on that the `token` parameter is not required. (In fact, if the session declares itself safe but the request still includes a token, an error is raised, which strikes me as overly hostile to API consumers – but whatever.)

But that code is currently in `EditHandler`, a subclass of `ActionModuleBasedHandler`, so we probably couldn’t directly use it even if we were targeting MediaWiki master (rather than 1.35).

Reply:
Ok, have a look at https://github.com/wmde/WikibaseReconcileEdit/pull/35/commits/2f1132640bf84050be6050160d0143894462899b

It switches these parameters from being passed in post but rather all together in the body. Made the tests and the code to look a bit more similar to what is being done in UpdateHandler.